process/interceptors/factory/interceptedHeartbeatDataFactory.go

```diff
@@ -0,0 +1,50 @@
+package factory
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/heartbeat"
+)
+
+type interceptedHeartbeatDataFactory struct {
+	marshalizer marshal.Marshalizer
+	peerID      core.PeerID
+}
+
+// NewInterceptedHeartbeatDataFactory creates an instance of interceptedHeartbeatDataFactory
+func NewInterceptedHeartbeatDataFactory(arg *ArgInterceptedDataFactory) (*interceptedHeartbeatDataFactory, error) {
+	if arg == nil {
```

Comment:
this can be removed if the arg is of type `ArgInterceptedDataFactory` instead of `*ArgInterceptedDataFactory`